spec/unit/artifactserver_spec.rb

```diff
@@ -56,6 +56,59 @@ module CookbookOmnifetch
       }
       expect(public_repo_location.lock_data).to eq(expected_data)
     end
+
+    context "when asked to install a new cookbook" do
+
+      let(:http_client) { double("Http Client") }
+
+      let(:test_root) { Dir.mktmpdir(nil) }
+
+      let(:storage_path) { File.join(test_root, "storage") }
+
+      let(:cache_path) { File.join(test_root, "cache") }
+
+      let(:path) { fixtures_path.join('cookbooks') }
+
+      let(:cookbook_name) { "example_cookbook" }
+
+      let(:cookbook_version) { "0.5.0" }
+
+      before do
+        allow(CookbookOmnifetch).to receive(:storage_path).and_return(Pathname.new(storage_path))
+        allow(CookbookOmnifetch).to receive(:cache_path).and_return(cache_path)
+        FileUtils.mkdir_p(storage_path)
+      end
+      it "installs the cookbook to the desired install path" do
+        expect(public_repo_location).to receive(:http_client).and_return(http_client)
+        expect(http_client).to receive(:streaming_request) do |arg, &block|
+          Dir.mktmpdir(nil) do |source_dir|
```

Comment:
This would be a lot cleaner if you either just add a new fixture with a tarball-ed cookbook, or move creation into a let block. Then you can do:

```ruby
# Test "spy", make sure we close it
expect(tarball_tempfile_filehandle).to receive(:close).and_call_original
# #and_yield(thing) passes thing as the block argument
expect(http_client).to receive(:streaming_request).with(nil).and_yield(tarball_tempfile_filehandle)
```